### PR TITLE
Feature/dashboard

### PR DESCRIPTION
**Pull request by Prakriti-Soni**

### CODE REVIEW

File: src/appRouter.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ import { getToast, getOverlay } from './js/store/common/selectors';
 import LoadingOverlay from './js/common/overlay';
 import GlobalStyles from './js/hooks/globalStyles';
```
Comment (jineshmehta16) [inline src/appRouter.js:9]: if you dont need global styles you can get rid of it

File: src/appRouter.js [hunk 2]
```diff
@@ -8,6 +8,7 @@ import { getToast, getOverlay } from './js/store/common/selectors';
...
 import SplashScreen from './js/component/splashScreen';
+import ParkingMap from './js/component/parkingMap';
```
Comment (jineshmehta16) [inline src/appRouter.js:11]: add lazy loading, code splitting will help a lot to improve page performance

File: src/js/component/header/index.js [hunk 3]
```diff
@@ -1,153 +1,157 @@
...
-            position='static'
-            color='default'
-            id='back-to-top-anchor'
-            className={`${'px-4 font-size-20px'} ${
-              classes?.headerWhiteBackground
-            }`}
+  const drawer = (
+    <Box onClick={handleDrawerToggle} sx={{ textAlign: 'center' }}>
+      <Typography variant='h6' sx={{ my: 2 }}>
+        HyP Parking Management Dashboard
```
Comment (jineshmehta16) [inline src/js/component/header/index.js:45]: try and put all static text in constant file

File: src/js/component/header/index.js [hunk 4]
```diff
@@ -1,153 +1,157 @@
...
+              aria-label='open drawer'
+              edge='start'
+              onClick={handleDrawerToggle}
+              sx={{ mr: 2, display: { sm: 'none' } }}
+            >
+              <MenuIcon />
+            </IconButton>
+          )}
+          <Typography variant='h6' component='div' sx={{ flexGrow: 1 }}>
+            HyP Parking Management Dashboard
```
Comment (jineshmehta16) [inline src/js/component/header/index.js:96]: leverage your constant file here

File: src/js/component/parkingMap/index.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    <>
+      <div id='map' className={classes.wrapper}>
+        <img src={state?.imagePath} alt='test' className={classes.image} />
+        {dataSet.map((each) => (
+          <div
+            key={each?.parkingNumber}
+            style={{
+              width: each?.position === 'H' ? '2%' : '1.3%',
+              height: each?.position === 'H' ? '11px' : '33px',
+              background: +parkingStatus?.[each?.parkingNumber]
```
Comment (jineshmehta16) [inline src/js/component/parkingMap/index.js:37]: 1 hoga to green dikhana hai na ?

File: src/js/component/reportForm/index.js [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+
+const ReportForm = (props) => {
+  const { classes } = props;
+  const dispatch = useDispatch();
+  const [reportFormat, setReportFormat] = useState('daily');
+  const [selectedDate, setSelectedDate] = useState(null);
+  const [selectedMonth, setSelectedMonth] = useState('');
+  const [selectedYear, setSelectedYear] = useState('');
+
+  const months = [
```
Comment (jineshmehta16) [inline src/js/component/reportForm/index.js:29]: better to put all those in one js file and import it in this component. Tomorrow you might need them in other component too

File: src/js/component/reportForm/index.js [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+        >
+          <FormControlLabel value='daily' control={<Radio />} label='Daily' />
+          <FormControlLabel
+            value='monthly'
+            control={<Radio />}
+            label='Monthly'
+          />
+        </RadioGroup>
+      </FormControl>
+      {reportFormat === 'daily' && (
```
Comment (jineshmehta16) [inline src/js/component/reportForm/index.js:104]: daily/ monthly ye sab static string ko constant file se uthaana sahi rahega for long term

File: src/js/component/reportForm/index.js [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+            <InputLabel id='demo-simple-select-label'>Year</InputLabel>
+            <Select
+              labelId='demo-simple-select-label'
+              id='demo-simple-select'
+              value={selectedYear}
+              label='Year'
+              onChange={(e) => setSelectedYear(e.target.value)}
+            >
+              {years?.map((year) => (
+                <MenuItem value={year}>{year}</MenuItem>
```
Comment (jineshmehta16) [inline src/js/component/reportForm/index.js:153]: key missing

Comment (jineshmehta16) [reply src/js/component/reportForm/index.js:153]: whenever you use map, always remember to put key..and do not put index as it value..

File: src/js/store/dashboard/action.js [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    payload: data,
+  };
+};
+
+export const getOverallParkingInformation = () => {
+  return (dispatch) => {
+    dispatch(setOverlayStatus(true));
+    return get('/occupancy')
+      .then((response) => {
+        if (response.data.status.toUpperCase() === 'SUCCESS') {
```
Comment (jineshmehta16) [inline src/js/store/dashboard/action.js:17]: ?. 

for checking undefined

File: src/js/store/dashboard/action.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+          dispatch(setOverallParkingDetails(response.data.data));
```
Comment (jineshmehta16) [inline src/js/store/dashboard/action.js:18]: ?.

**Review by jineshmehta16 — CHANGES_REQUESTED** [review]
some minor changes

File: src/appRouter.js (hunk 1 above)
Comment (Prakriti-Soni) [reply src/appRouter.js:9]: removed

File: src/js/component/header/index.js (hunk 4 above)
Comment (Prakriti-Soni) [reply src/js/component/header/index.js:96]: done

File: src/js/component/reportForm/index.js (hunk 7 above)
Comment (Prakriti-Soni) [reply src/js/component/reportForm/index.js:104]: done

File: src/js/axiosUtils/appUtils.js [hunk 11]
```diff
@@ -1,79 +1,39 @@
...
 import handleError from './handleError';
-import handleGraphQlError from './handleGraphQlError';
+
 const baseUrl = `${APP_CONFIG.API_URL}`;
-const baseImageUrl = `${APP_CONFIG.IMAGE_URL}`;
-const nodeBaseUrl = `${APP_CONFIG.NODE_URL}`;
 
 function getAccessToken() {
-  return localStorage.getItem('accessToken');
+  return sessionStorage.getItem('token');
```
[the remark this answers is not in the document]
Comment (jineshmehta16) [reply src/js/axiosUtils/appUtils.js:8]: if someone refresh the page token will not be there if you store it in sesion

File: src/js/component/parkingMap/index.js [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+
+const ParkingMap = (props) => {
+  const { classes } = props;
+  const { state } = useLocation();
+  const [dataSet, setDataset] = useState([]);
+  const [parkingStatus, setParkingStatus] = useState({});
+  const refreshFlag = useSelector((state) => getRefreshedPageData(state));
+
+  useEffect(() => {
+    axios.get('/db/parkingMapData.json').then((response) => {
```
Comment (jineshmehta16) [inline src/js/component/parkingMap/index.js:18]: We should have dispatch things from this layer to action and waha pe we can call our apputil.js which already has axios layer right ?

Comment (jineshmehta16) [reply src/js/component/parkingMap/index.js:18]: I am sure there is some reason aap ne ye keya hoga ?

**Review by jineshmehta16 — CHANGES_REQUESTED** [review]
minor

File: src/js/component/parkingMap/index.js (hunk 12 above)
Comment (Prakriti-Soni) [reply src/js/component/parkingMap/index.js:18]: we discussed

File: src/js/component/parkingMap/index.js (hunk 5 above)
Comment (Prakriti-Soni) [reply src/js/component/parkingMap/index.js:39]: yes it will be green

**Pull request merged**